universe/auto_syncer.go

```diff
@@ -468,10 +468,60 @@ func (f *FederationEnvoy) SyncServers(serverAddrs []ServerAddr) error {
 	ctx, cancel := f.WithCtxQuitNoTimeout()
 	defer cancel()
 
-	err := fn.ParSlice(ctx, serverAddrs, f.syncServerState)
+	// Obtain the general and universe specific federation sync configs.
+	queryFedSyncConfigs := f.cfg.FederationDB.QueryFederationSyncConfigs
+	globalConfigs, uniSyncConfigs, err := queryFedSyncConfigs(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to query federation sync "+
+			"config(s): %w", err)
+	}
+
+	syncConfigs := SyncConfigs{
+		GlobalSyncConfigs: globalConfigs,
+		UniSyncConfigs:    uniSyncConfigs,
+	}
+
+	syncServer := func(ctx context.Context, serverAddr ServerAddr) error {
+		return f.syncServerState(ctx, serverAddr, syncConfigs)
+	}
+
+	err = fn.ParSlice(ctx, serverAddrs, syncServer)
 	if err != nil {
 		log.Warnf("unable to sync with server: %w", err)
 	}
 
 	return nil
 }
+
+// SyncConfigs is a set of configs that are used to control which universes to
+// synchronize with the federation.
+type SyncConfigs struct {
+	// GlobalSyncConfigs are the global proof type specific configs.
+	GlobalSyncConfigs []*FedGlobalSyncConfig
+
+	// UniSyncConfigs are the universe specific configs.
+	UniSyncConfigs []*FedUniSyncConfig
+}
+
+// IsSyncEnabled returns true if the given universe is configured to be
+// synchronized with the federation.
+//
+// TODO(ffranr): Add support for insert/export only sync enabled status.
+func (s *SyncConfigs) IsSyncEnabled(id Identifier) bool {
+	// Check for universe specific config. This takes precedence over the
+	// global config.
+	for _, cfg := range s.UniSyncConfigs {
+		if cfg.UniverseID == id {
+			return cfg.AllowSyncInsert && cfg.AllowSyncExport
```

Comment:
I think the export should be eliminated, as we only need to check that when we go to process RPC requests.